packages/features/eventtypes/components/EditWeightsForAllTeamMembers.tsx

```diff
@@ -0,0 +1,304 @@
+                  const newWeight = parseInt(e.target.value);
+                  if (!isNaN(newWeight)) {
+                    onWeightChange(member.value, newWeight);
+                  }
+                  setIsEditing(false);
+                }}
+                onKeyDown={(e) => {
+                  if (e.key === "Enter") {
+                    const input = e.target as HTMLInputElement;
+                    const newWeight = parseInt(input.value);
+                    if (!isNaN(newWeight)) {
+                      onWeightChange(member.value, newWeight);
+                    }
+                    setIsEditing(false);
+                  }
+                  if (e.key === "Escape") {
+                    setIsEditing(false);
+                  }
+                }}
+              />
+              <span className="text-default border-default bg-muted flex h-7 items-center rounded-r-sm border border-l-0 px-2 text-sm">
+                %
+              </span>
+            </div>
+          </div>
+        ) : (
+          <button
+            className="text-emphasis hover:bg-subtle decoration-emphasis flex h-7 items-center rounded-sm px-2 text-sm underline underline-offset-4"
+            onClick={() => setIsEditing(true)}>
+            {member.weight ?? 100}%
+          </button>
+        )}
+      </div>
+    </div>
+  );
+};
+
+interface Props {
+  teamMembers: TeamMember[];
+  value: Host[];
+  onChange: (hosts: Host[]) => void;
+}
+
+export const EditWeightsForAllTeamMembers = ({ teamMembers, value, onChange }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { t } = useLocale();
+  const [searchQuery, setSearchQuery] = useState("");
+  const [localWeights, setLocalWeights] = useState<Record<string, number>>(() =>
+    teamMembers.reduce<Record<string, number>>((acc, member) => {
+      // Find the member in the value array and use its weight if it exists
+      const memberInValue = value.find((host) => host.userId === parseInt(member.value, 10));
+      acc[member.value] = memberInValue?.weight ?? 100;
+      return acc;
+    }, {})
+  );
+  const [uploadErrors, setUploadErrors] = useState<Array<{ email: string; error: string }>>([]);
+  const [isErrorsExpanded, setIsErrorsExpanded] = useState(true);
+
+  const handleWeightChange = (memberId: string, weight: number) => {
+    setLocalWeights((prev) => ({ ...prev, [memberId]: weight }));
+  };
+
+  const handleSave = () => {
+    const updatedValue = value.map((host) => ({
+      ...host,
+      weight: localWeights[host.userId.toString()] ?? host.weight ?? 100,
+    }));
+    onChange(updatedValue);
+    setIsOpen(false);
+    showToast(t("weights_updated_successfully"), "success");
+  };
+
+  const handleDownloadCsv = () => {
+    const csvData = teamMembers.map((member) => ({
+      id: member.value,
+      name: member.label,
+      email: member.email,
+      weight: member.weight ?? 100,
+    }));
+    downloadAsCsv(csvData, "team-members-weights.csv");
```

Comment:
my downloaded csv shows weight 100 for all hosts even though I have some other weights